lib/matplotlib/artist.py

```diff
@@ -594,11 +594,26 @@ def set_path_effects(self, path_effects):
     def get_path_effects(self):
         return self._path_effects
 
+    @property
+    def figure(self):
+        """:class:`~matplotlib.figure.Figure` instance the artist
+        belongs to"""
+        return self._figure
+
+    @figure.setter
+    def figure(self, fig):
+        self._figure = fig
```

Comment:
If you are going to do this you should add validation that artists can not be moved between figures.